content/guides/code-splitting.md

```diff
@@ -1,41 +1,290 @@
+Code splitting is one of the most compelling features of webpack. This feature allows you to split your code into various bundles which can then be loaded on demand or in parallel. It can be used to achieve smaller bundles and control resource load prioritization which, if used correctly, can have a major impact on load time.
 
+There are three general approaches to code splitting available:
 
-## Resource Splitting
+- Entry Points: Manually split code using [`entry`](/configuration/entry-context) configuration.
+- Prevent Duplication: Use the [`CommonsChunkPlugin`](/plugins/commons-chunk-plugin) to dedupe and split chunks.
+- Dynamic Imports: Split code via inline function calls within modules.
 
-Vendor and CSS code splitting methods help with both caching and parallel loading.
 
-### Vendor Code Splitting
+## Entry Points
 
-A typical application can depend on many third party libraries for framework/functionality needs. Unlike application code, code present in these libraries does not change often.
+This is by far the easiest, and most intuitive, way to split code. However, it is more manual and has a some pitfalls we will go over. Let's take a look at how we might split another module from the main bundle:
 
-If we keep code from these libraries in its own bundle, separate from the application code, we can leverage the browser's caching mechanism to cache these files for longer durations on the end user's machine.
+__project__
 
-For this to work, the `[hash]` portion in the vendor filename must remain constant, regardless of application code changes. Learn [how to split vendor/library](/guides/code-splitting-libraries) code using the `CommonsChunkPlugin`.
+``` diff
+webpack-demo
+|- package.json
+|- webpack.config.js
+|- /dist
+|- /src
+  |- index.js
++ |- another-module.js
+|- /node_modules
+```
 
-### CSS Splitting
+__another-module.js__
 
-You might also want to split your styles into a separate bundle, independent from application logic.
-This enhances cacheability of your styles and allows the browser to load the styles in-parallel with your application code, thus preventing a FOUC ([flash of unstyled content](https://en.wikipedia.org/wiki/Flash_of_unstyled_content)).
+``` js
+import _ from 'lodash';
 
-Learn [how to split CSS](/guides/code-splitting-css) using the `ExtractTextWebpackPlugin`.
+console.log(
+  _.join(['Another', 'module', 'loaded!'], ' ')
+);
+```
 
+__webpack.config.js__
 
-## On Demand Code Splitting
+``` js
+const path = require('path');
+const HTMLWebpackPlugin = require('html-webpack-plugin');
 
-While resource splitting of the previous kind requires the user to specify the split points upfront in the configuration, one can also create dynamic split points in the application code.
+module.exports = {
+  entry: {
+    index: './src/index.js',
+    another: './src/another-module.js'
+  },
+  output: {
+    filename: '[name].bundle.js',
+    path: path.resolve(__dirname, 'dist')
+  },
+  plugins: [
```

Comment:
Minor: In other examples we put `plugins` before `output`. I don't care much about the order of things, but it might be nice to keep it consistent.